src/main/kotlin/org/danilopianini/plagiarismdetector/analyzer/technique/tokenization/java/JavaParser.kt

```diff
@@ -0,0 +1,18 @@
+package org.danilopianini.plagiarismdetector.analyzer.technique.tokenization.java
+
+import com.github.javaparser.ParserConfiguration
+import com.github.javaparser.StaticJavaParser
+import com.github.javaparser.ast.CompilationUnit
+import org.danilopianini.plagiarismdetector.analyzer.StepHandler
+import java.io.File
+
+/**
+ * A java source file parser.
+ */
+class JavaParser : StepHandler<File, CompilationUnit> {
+    override fun invoke(input: File): CompilationUnit {
+        val parserConfiguration = ParserConfiguration().setAttributeComments(false)
+        StaticJavaParser.setConfiguration(parserConfiguration)
```

Comment:
It looks like you have a mutable shared `object`. This is extremely fragile and should not happen.